terraform-aws-modules/terraform-aws-vpc
# feat: add default route table resource to manage default route table, its tags, routes, etc.

## Description
- add default route table resource to manage default route table, its tags, routes, etc.
- update complete vpc example to show/validate changes here for managing the default route table 
- `vpc_endpoint_id` was left out of the `route` config block because it was only recently added in v3.15.0 of the AWS provider and it will throw errors if the min version is not set to that. we can re-enable this later on when the min version bump is justified

## Motivation and Context
- adds the ability to tag and manage the default (`main`) route table of the VPC created

These changes supersedes the following PRs:
Closes #545 
Closes #147

## Breaking Changes
- No

## How Has This Been Tested?
- Checkout `master` locally and ensure its up to date/current
- `cd examples/complete-vpc && terraform init && terraform apply` -> sets baseline
- Checkout these changes locally
- `terraform apply` -> manages default route table and sets the tags, etc.

Note - I also tested going from `master` to this change set without any changes added to the example project and the plan came back clean meaning backwards compatibility is preserved
